crates/core/src/pattern/resolved_pattern.rs

```diff
@@ -98,48 +98,22 @@ impl<'a> File<'a> {
     }
 }
 
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) struct ListBinding<'a> {
-    pub(crate) src: &'a str,
-    pub(crate) parent_node: Node<'a>,
-    pub(crate) field: FieldId,
-}
-
-impl<'a> ListBinding<'a> {
-    pub(crate) fn new(src: &'a str, parent_node: Node<'a>, field: FieldId) -> Self {
-        Self {
-            src,
-            parent_node,
-            field,
-        }
-    }
-}
-
-#[derive(Debug, Clone, PartialEq)]
-pub(crate) enum Lists<'a> {
-    Binding(ListBinding<'a>),
-    Resolved(Vector<ResolvedPattern<'a>>),
-}
-
 #[derive(Debug, Clone, PartialEq)]
 pub struct JoinFn<'a> {
-    pub(crate) list: Lists<'a>,
+    pub(crate) list: Vector<ResolvedPattern<'a>>,
```

Comment:
Is this what you referring to where you had to eagerly evaluate? I think this is fine. The purpose of the function structs is to be able to lazily evaluate the text the nodes are pointing to. Specifically if a function is on the right hand side of a rewrite then the function will be evaluated at linearization time. Sometimes the functions are eagerly evaluated eg. the text function eagerly evaluates the enclosed resolved pattern.

Reply:
Thanks! This is the one indeed.